### PR TITLE
ODC-6694: New service binding plugin with a list and detail page, updated topology package

### PR DESCRIPTION
**Pull request by jerolimov**
**Fixes**: 
https://issues.redhat.com/browse/ODC-6694

**Analysis / Root cause**: 
As a user, I want to understand which service bindings connected service to a component successfully or not. Currently, it's really difficult to understand and needs inspection into each ServiceBinding resource (yaml).

**Solution Description**: 
This is the first half of the requirements and shows the status of a Service Binding in the list and detail view.

Will open another PR for the topology view and sidebar!

**Screen shots / Gifs for design review**: 

Search result or pinned service binding list:

image

When clicking on an error:

image

Service binding details page:

image

New topology status (one SB is failed, the other one is fine)

https://user-images.githubusercontent.com/139310/172696422-66126a6c-e294-4047-b0a7-a37b1c9643a6.mp4

https://user-images.githubusercontent.com/139310/172696470-459458bd-1a18-4bad-92ba-7ee526262317.mp4

When selected the service binding (graph view is outdated !) the status is also visible in the sidebar:

image

**Unit test coverage report**: 
Added unit tests for the detail and list views.

**Test setup:**
1. Install Service Binding operator
2. Install a operator that supports SBO, like Progresql operator or Redis operator
3. Create or switch namespace if needed
4. Import from git or container
5. Navigate to Add > Operator Backed and select "Bindable" on the left side
6. Install "Redis"
8. Drag and drop from your Deployment to the Redis StatefulSet

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

/cc @invincibleJai @divyanshiGupta

### CODE REVIEW

File: frontend/public/components/utils/resource-link.tsx [hunk 1]
```diff
@@ -6,20 +6,18 @@ import * as classNames from 'classnames';
...
-  referenceForModel,
-  K8sKind,
-  K8sResourceKindReference,
-  K8sResourceKind,
-} from '../../module/k8s';
+import { K8sModel } from '@console/dynamic-plugin-sdk/src/api/common-types';
+import { K8sResourceKindReference } from '@console/dynamic-plugin-sdk/src/extensions/console-types';
+import { K8sResourceKind } from '../../module/k8s/types';
+import { modelFor } from '../../module/k8s/k8s-models';
+import { referenceForModel } from '../../module/k8s/k8s-ref';
```
Comment (jerolimov) [inline frontend/public/components/utils/resource-link.tsx:13]: fyi: This update was needed to write the unit tests with less mocks.

**Comment by jerolimov** [comment]
@invincibleJai @divyanshiGupta I know there is another SB modal in `dev-console`, but I will create a tech debt issue for extracting move code into this new plugin. Maybe I will do this already with the 2nd PR.

**Comment by jerolimov** [comment]
Propagate from the Epic
/label docs-approved
/label px-approved

File: frontend/packages/service-binding-plugin/console-extensions.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+[
+  {
```
[the remark this answers is not in the document]
Comment (jerolimov) [reply frontend/packages/service-binding-plugin/console-extensions.json:2]: added

File: frontend/packages/service-binding-plugin/src/components/service-binding-details/ServiceBindingDetailsTab.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+              <DetailsItem
+                label={t('service-binding-plugin~Services')}
+                obj={serviceBinding}
+                path="spec.services"
+              >
+                {serviceBinding.spec?.services?.length
+                  ? serviceBinding.spec.services.map((service, index) => (
+                      <ResourceLink
+                        // eslint-disable-next-line react/no-array-index-key
+                        key={index}
```
[the remark this answers is not in the document]
Comment (jerolimov) [reply frontend/packages/service-binding-plugin/src/components/service-binding-details/ServiceBindingDetailsTab.tsx:55]: Evil people can also use the same data twice:

image

Let us keep this simple and use the index, ok?

**Comment by invincibleJai** [comment]
Looks good to me and works as mentioned, need to run `yarn i18n` for missing translations.

**Comment by vikram-raj** [comment]
Verified this and everything works as expected.

**Success scenarios**
image

image

image

**Error Scenarios**
image

image

image

**Review by vikram-raj — COMMENTED** [review]
/label qe-approved

**Comment by invincibleJai** [comment]
/lgtm

verified it, and works as expected. Thanks 👍 

image

**Comment by jhadvig** [comment]
```
info There appears to be trouble with your network connection. Retrying...
error An unexpected error occurred: "https://registry.yarn...y/react-tokens/-/react-tokens-4.12.5.tgz: ESOCKETTIMEDOUT".
```
🙌

**Pull request merged**